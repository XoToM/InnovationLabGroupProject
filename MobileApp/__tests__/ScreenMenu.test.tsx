import React from "react";
import { render } from "@testing-library/react-native";
import ScreensMenu from "../app/screens/menu";

// Mock the expo-router's Link component
jest.mock("expo-router", () => ({
  Link: ({ children }: any) => <>{children}</>,
}));

describe("ScreensMenu", () => {
  it("renders the screen title", () => {
    const { getByText } = render(<ScreensMenu />);
    expect(getByText("Screens Menu")).toBeTruthy();
  });

  it("renders the Login button", () => {
    const { getByText } = render(<ScreensMenu />);
    expect(getByText("Login")).toBeTruthy();
  });

  it("renders the SignUP button", () => {
    const { getByText } = render(<ScreensMenu />);
    expect(getByText("SignUP")).toBeTruthy();
  });

  it("contains buttons wrapped in Link", () => {
    const { getByText } = render(<ScreensMenu />);
    expect(getByText("Login")).toBeTruthy();
    expect(getByText("SignUP")).toBeTruthy();
  });

  it("matches the snapshot", () => {
    const tree = render(<ScreensMenu />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
