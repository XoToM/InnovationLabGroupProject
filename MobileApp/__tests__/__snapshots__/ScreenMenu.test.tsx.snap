// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreensMenu matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Screens Menu
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        SignUP
      </Text>
    </View>
  </View>
</View>
`;
